src/app/pages/network/interfaces/interfaces-list/interfaces-list.component.ts

```diff
@@ -1,55 +1,146 @@
+    {name : T('IP Addresses'), prop : 'addresses'}
   ];
   public config: any = {
     paging : true,
     sorting : {columns : this.columns},
     deleteMsg: {
       title: 'Interface',
-      key_props: ['int_interface']
+      key_props: ['name']
     },
   };
 
-  constructor(_rest: RestService, private router: Router, private networkService: NetworkService,
+  constructor(private ws: WebSocketService, private router: Router, private networkService: NetworkService,
               private snackBar: MatSnackBar) {}
 
-  rowValue(row, attr) {
-    if (attr == 'ipv4_addresses' || attr == 'ipv6_addresses') {
-      return row[attr].join(', ');
+  dataHandler(res) {
+    const rows = res.rows;
+    for (let i=0; i<rows.length; i++) {
+      rows[i]['link_state'] = rows[i]['state']['link_state'];
+      const addresses = [];
+      for (let j=0; j<rows[i]['aliases'].length; j++) {
+        const alias = rows[i]['aliases'][j];
+        if (alias.type.startsWith('INET')) {
+          addresses.push(alias.address);
+        }
+      }
+      rows[i]['addresses'] = addresses.join(', ');
+      rows[i].details = []
+      if (rows[i].type === "PHYSICAL") {
+        rows[i].details.push({label: T("Active Media Type"), value:rows[i]["state"]["active_media_type"]},
+                             {label: T("Active Media Subtype"), value:rows[i]["state"]["active_media_subtype"]})
+      } else if (rows[i].type === "VLAN") {
+        rows[i].details.push({label: T("VLAN Tag"), value:rows[i]["vlan_tag"]},
+                             {label: T("VLAN Parent Interface"), value: rows[i]["state"]["vlan_parent_interface"]})
+      } else if (rows[i].type === "BRIDGE") {
+        rows[i].details.push({label:T("Bridge Members"), value:rows[i]["bridge_members"]});
+      } else if (rows[i].type === "LINK_AGGREGATION") {
+        rows[i].details.push({label:T("Lagg Ports"), value:rows[i]["lag_ports"]},
+                             {label:T("Lagg Protocol"), value:rows[i]["lag_protocol"]});
+      }
+      rows[i].details.push({label:T("IP Addresses"), value:rows[i]['addresses']});
+      rows[i].details.push({label:T("MAC Address"), value:rows[i]['state']['link_address']});
     }
-    return row[attr];
+
+  }
+
+  preInit(entityList) {
+    this.entityList = entityList;
+
+    this.checkPendingChanges();
+    this.checkChangesSubscription = interval(10000).subscribe(x => {
+      this.checkPendingChanges();
+    });
+  }
+
+  checkPendingChanges() {
+    this.ws.call('interface.has_pending_changes').subscribe(res => {
+      this.hasPendingChanges = res;
+    });
+  }
+
+  commitPendingChanges() {
+    this.entityList.dialogService.confirm(
+      helptext.commit_changes_title,
+      helptext.commit_changes_warning,
+      false, helptext.commit_button).subscribe(confirm => {
+        if (confirm) {
+          this.entityList.loader.open();
+          this.entityList.loaderOpen = true;
+          this.ws.call('interface.commit').subscribe(res => {
```

Comment:
That method has a `checkin_timeout` option, it defaults to 60 seconds. It means if the user doesnt come back confirming the changes within that period it will get automatically rolled back.

We should mention that, or make it configurable.

Reply:
I will add that :)